### PR TITLE
Enable capturing of stdout/stderr as an eventlistener

### PR DESCRIPTION
**Pull request by jib**
This allows stdout/stderr events to be captured like this:

```
{
  "groupname": "logging_test",
  "tags": [],
  "@version": "1",
  "@timestamp": "2016-03-29T00:21:53.789Z",
  "level": "INFO",
  "pid": "8644",
  "eventname": "PROCESS_LOG_STDERR",
  "processname": "logging_test",
  "host": "ip-10-93-130-24",
  "logger_name": "supervisor",
  "path": "/home/ubuntu/sources/git/supervisor-logstash-notifier/logstash_notifier/__init__.py",
  "message": "Tue Mar 29 00:21:53 2016 at -e line 1.\n",
  "type": "logstash",
  "channel": "stderr"
}
{
  "groupname": "logging_test",
  "tags": [],
  "@version": "1",
  "@timestamp": "2016-03-29T00:21:58.789Z",
  "level": "INFO",
  "pid": "8644",
  "eventname": "PROCESS_LOG_STDOUT",
  "processname": "logging_test",
  "host": "ip-10-93-130-24",
  "logger_name": "supervisor",
  "path": "/home/ubuntu/sources/git/supervisor-logstash-notifier/logstash_notifier/__init__.py",
  "message": "Tue Mar 29 00:21:58 2016\n",
  "type": "logstash",
  "channel": "stdout"
}
```

### CODE REVIEW

File: logstash_notifier/__init__.py [hunk 1]
```diff
@@ -148,10 +152,10 @@ def application(include=None):
             if len(user_data) > 0:
                 extra['user_data'] = user_data
 
-        logger.info(
-            '%s %s', headers['eventname'], event_body['processname'],
-            extra=extra,
-        )
+        message = event_data if len(event_data) \
+                             else '%s %s' % (headers['eventname'], event_body['processname'])
```
Comment (hanpeter) [inline logstash_notifier/__init__.py:160]: Instead of replacing the existing logging, don't you rather add this logging?

File: tests/test_logstash_notifier.py [hunk 2]
```diff
@@ -205,6 +205,48 @@ def test_get_user_data(self):
...
             )
 
+class SupervisorOutPutLoggingTestCase(BaseSupervisorTestCase):
+    """
+    Test capturing stdout/stderr logs.
+    """
+
+    def test_output_logging(self):
+        """
+        Test output logging.
```
Comment (hanpeter) [inline tests/test_logstash_notifier.py]: This docstring adds no value. How about something like "Test stdout is captured in logs when capture-output argument is set"

Comment (jib) [inline tests/test_logstash_notifier.py]: Thanks, done.

**Comment by hanpeter** [comment]
Also, one more thing: You probably want a test where `event_data` variable in `application()` function is empty. Just to check that the existing logging still happens even with the new arguments set.

File: logstash_notifier/__init__.py (hunk 1 above)
Comment (jib) [inline logstash_notifier/__init__.py:160]: No; events don't have a message body (event_data), so its set to '' higher up and overridden here with here with the '%s %s' code. 

Stdout/Stderr output DO have a body (obviously), so we use that if it's present.

Added that as a comment for clarity.

**Comment by jib** [comment]
@hanpeter how would i do that in the testing framework? that seems buried pretty far down and not sure how to get to that.
